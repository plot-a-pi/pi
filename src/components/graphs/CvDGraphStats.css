.CvDGraphStats{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: .25em;
  margin: 1em 0 1em 0;
  width: 90vw;
  height: 25vw;
  border-radius: .5em;
  text-align: center;
  
  .card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-radius: 100%;
    background: rgb(197, 209, 240);
    height: 100%;
    width: 25vw;
  }

  h2 {
    font-size: 3vw;
    font-weight: bold;
    height: 4vw;
    margin-bottom: 2vw;
  }

  h3 {
    font-size: 5vw;
  }

  .formula {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-size: 90%;
    height: 100%;
  }
}
