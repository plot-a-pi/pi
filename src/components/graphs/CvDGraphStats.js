import React from 'react';
import MathJax from 'react-mathjax';
import styles from './CircumferenceVsDiameterGraph.css';
import PropTypes from 'prop-types';

const CvDGraphStats = ({ stats }) => {

  let statsEquation = '\\pi \\, \\approx \\, \\frac {}{}';

  if(!stats) return;
  else {statsEquation = `\\pi \\, \\approx \\, \\frac {c}{d} \\, \\approx \\, ${stats.mean}`;}
  
  
  return (
    <div className={styles.stats}>
      <h2>Total Points: <span>{stats.count}</span></h2>
      <MathJax.Provider>
        <div className={styles.stats}>
          <MathJax.Node formula={statsEquation} />
        </div>
      </MathJax.Provider>
    </div>

  );
};

CvDGraphStats.propTypes = {
  stats: PropTypes.object.isRequired
};

export default CvDGraphStats;
