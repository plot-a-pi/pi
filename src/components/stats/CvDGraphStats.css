.CvDGraphStats{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 22vw;
  margin-top: 3vw;
  margin-bottom: 3vw;
  border-radius: .5em;
  text-align: center;
  
  .card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-radius: 100%;
    background: rgb(197, 209, 240);
    height: 100%;
    width: 22vw;
    
  }

  h2 {
    font-size: 3vw;
    font-weight: bold;
    height: 4vw;
    margin-bottom: 2vw;
  }

  h3 {
    font-size: 5vw;
  }

  .formula {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-size: 90%;
    height: 100%;
  }

  @media screen and (min-width: 1200px) {
    height: 10vw;
    justify-content: space-around;

    .card {
      width: 10vw;
      justify-content: center;
    }

    h2 {
      font-size: 1.5vw;
      height: 2vw;
      margin-bottom: 0.5vw;
    }

    h3 {
      font-size: 2.2vw;
    }
}
}
