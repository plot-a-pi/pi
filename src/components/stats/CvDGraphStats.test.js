import React from 'react';
import { shallow } from 'enzyme';
import CvDGraphStats from './CvDGraphStats';

describe('CvDGraphStats component', () => {
  it('renders the CvDGraphStats', () => {
    const wrapper = shallow(<CvDGraphStats stats={{ mean: 3, count: 10 }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
