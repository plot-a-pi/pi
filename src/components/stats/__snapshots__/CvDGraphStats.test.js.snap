// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvDGraphStats component renders the CvDGraphStats 1`] = `
<div
  className="CvDGraphStats"
>
  <div
    className="card"
  >
    <h2>
      Points
    </h2>
    <h3>
      10
    </h3>
  </div>
  <div
    className="card"
  >
    <h2>
      π ≈
    </h2>
    <h3>
      3.00
    </h3>
  </div>
  <div
    className="card"
  >
    <h2>
      Error
    </h2>
    <h3>
      4.51
      %
    </h3>
  </div>
</div>
`;
